### DISCUSSION TITLE
Flush conn logs

### DISCUSSION THREAD

**Discussion by ketulsheth2** [zeek/zeek#1414] [Help]
Hi, I am using tcpreplay to send small pcap files through zeek and looking for a way to flush conn logs as soon as it sees a new connection. Is there a way to disable buffering logs?

**Comment by jsiwek** [comment]
Not a way to outright disable log buffering, but could try to either reduce the batching/buffering interval:

https://github.com/zeek/zeek/blob/7b4b8bd96d9c84a0468b19c7b69e4ef0e6bba282/scripts/base/frameworks/broker/main.zeek#L70

Or explicitly call `Broker::flush_logs()` at a point of your choosing:

https://github.com/zeek/zeek/blob/7b4b8bd96d9c84a0468b19c7b69e4ef0e6bba282/scripts/base/frameworks/broker/main.zeek#L306

Like if the idea is to minimize latency of `conn.log`, could try calling `Broker::flush_logs()` in the `Conn::log_conn()` event handler.  There's an example of using logging events here:

https://docs.zeek.org/en/master/frameworks/logging.html#define-a-logging-event

**Comment by JustinAzoff** [comment]
You might also be running into these options:

```
## If a TCP connection is inactive, time it out after this interval. If 0 secs,                              
## then don't time it out.                                                                                   
##                                                                                                           
## .. zeek:see:: udp_inactivity_timeout icmp_inactivity_timeout set_inactivity_timeout                       
const tcp_inactivity_timeout = 5 min &redef;                                                                 

## If a UDP flow is inactive, time it out after this interval. If 0 secs, then                               
## don't time it out.                                                                                        
##                                                                                                           
## .. zeek:see:: tcp_inactivity_timeout icmp_inactivity_timeout set_inactivity_timeout                       
const udp_inactivity_timeout = 1 min &redef;                                                                 
```

if your pcap files are partial connections, and missing things like fin or rst packets, zeek will take a few minutes to time out the connection.  Also if you do not use the --unique-ip option to tcpreplay, it will just use the same 5 tuple which will cause zeek to keep the connection active.

adding a script that has

```
redef tcp_inactivity_timeout=30secs;
redef udp_inactivity_timeout=30secs;
```

may make things work better for you.

**Comment by sethhall** [comment]
Are you asking about why you aren't getting any logs entries when a new connection is observed?  The conn log is only written out once a connection is closed and I don't think anyone has written a script to let you know when a new connection is started.
